### PR TITLE
Add discrete feature groups to COMPAS dataset

### PR DESCRIPTION
**Pull request by thomkeh**

### CODE REVIEW

File: ethicml/data/util.py [hunk 1]
```diff
@@ -92,3 +91,11 @@ def _first_segment(feature_name: str) -> str:
         start_idx += len_group
 
     return feature_slices
+
+
+def flatten_dict(dictionary: Mapping[str, List[str]]) -> List[str]:
+    """Flatten a dictionary of lists by joining all lists to one big list."""
+    flattened: List[str] = []
+    for group in dictionary.values():
```
Comment (MylesBartlett) [inline ethicml/data/util.py:99]:
```suggestion
    return list(dictionary.values())
```

Comment (thomkeh) [reply ethicml/data/util.py:99]: this doesn't flatten the lists

Comment (thomkeh) [reply ethicml/data/util.py:99]: there might be a better way to do this, but it's not that easy

Comment (MylesBartlett) [reply ethicml/data/util.py:99]: Ah yes, you're right

Comment (MylesBartlett) [reply ethicml/data/util.py:99]:
```suggestion
    return [feat for group in dictionary.values() for feat in group]
```

Comment (thomkeh) [reply ethicml/data/util.py]: done

**Pull request merged**